lib/gui/components/modal/services/modal.js

```diff
@@ -56,18 +56,32 @@ module.exports = function($uibModal, $q) {
       close: modal.close,
       result: $q((resolve, reject) => {
         modal.result
-          .then(resolve)
-          .catch((error) => {
+          .then((value) => {
+            AnalyticsService.logEvent('Modal closed', {
```

Comment:
I guess we can keep the "Modal closed" but simply add a new "Set dangerous setting" one and that should do it!